Dockerfile

```diff
@@ -1,11 +1,28 @@
-FROM golang:1.18 as workspace
-WORKDIR /go/src/github.com/testcontainers/moby-ryuk
+# -----------
+# Build Image
+# -----------
+FROM golang:1.18-alpine3.16 as build
+
+# Go build env
+ENV CGO_ENABLED=0
+
+WORKDIR /app
+
+# Install source deps
 COPY go.mod go.sum ./
 RUN go mod download
-COPY . ./
-RUN make build
 
-FROM alpine:3.13.6
+# Copy source & build
+COPY . .
+RUN go build -v -ldflags '-s' -o /bin/ryuk
```

Comment:
I noticed here and [here](https://github.com/testcontainers/moby-ryuk/pull/50/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR19-R30) that the existing make file is not used any more in favor of the go commands.

One thing that I expect from a build system such as Make, and from a CI, is that the CI must use the very same commands as the build system in order to provide developers a way for reproducing a CI build error.

In this PR we are removing the usages of the Makefile in favor of go test, which is fine to me at the moment we are making them explicit in the GH action and any developer would be able to read the GH workflow file.

On the other hand, Make provides an abstraction of the commands which is handy for developers to run the build system.

Which option do we prefer: Make Vs go commands? I would prefer having Make as entry point to the build, but I do not have a strong opinion on directly using go commands, producing a workflow with a simple set of steps.

Wdyt?

Reply:
Being more of an outsider I never really understood why Go projects use Make in an extremely simplistic fashion, often having a 1:1 mapping between goal and go command execution. However, if Make tends to be the default build system in the Go community it's of course good to follow this as a best practice and then I agree with @mdelapenya, that it should we used in all corresponding places to create artifacts.

Reply:
I'm not seeing Make as a default build system, but as best effort to define a proper build life cycle: compile > test > package as a minimum, probably biased by my previous experience with bigger Go projects.

I'm OK with removing it from the project, or keeping it, but being aware of the implications of it.

Reply:
I find make commands which are 1:1 mappings of the go commands are redundant unless standardized between repositories. I removed it from the Dockerfile to reduce the build cycle and explicitly define what's used inside the container. The dockerfile was a bit confusing since the context switched between cli, dockerfile, make stretching the context between three areas in a very short space:

* docker build -t testcontainers/ryuk
* Step into Dockerfile context 
* Step into Make context 
* Step back to Docker context

I can revert this change if you'd like

Reply:
I'm fine removing Make :)